scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -86,159 +94,182 @@ fun hasOpenGl31(context: Context): Boolean {
-        Rect(
-            0,
-            0,
-            toBorderWidth,
-            toBorderWidth
-        ),
-        Rect(
-            toBorderWidth,
-            0,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            0,
-            (2 * toBorderWidth + toCenterDimension),
-            toBorderWidth
-        ),
-        Rect(
-            0,
-            toBorderWidth,
-            toBorderWidth,
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            toBorderWidth,
-            toBorderWidth,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            112,
-            (2 * toBorderWidth + toCenterDimension),
-            toBorderWidth + toCenterDimension
-        ),
-        Rect(
-            0,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth,
-            (2 * toBorderWidth + toCenterDimension)
-        ),
-        Rect(
-            toBorderWidth,
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension,
-            (2 * toBorderWidth + toCenterDimension)
-        ),
-        Rect(
-            toBorderWidth + toCenterDimension,
-            toBorderWidth + toCenterDimension,
-            (2 * toBorderWidth + toCenterDimension),
-            (2 * toBorderWidth + toCenterDimension)
-        )
+    // Transforms the viewfinder from the preview to the image size
+    val projectedViewFinder = previewSize.projectRegionOfInterest(toSize = this.size(), regionOfInterest = cardFinder)
+    // Creates a square version of the viewfinder
+    val projectedViewFinderSquare = Size(projectedViewFinder.width(), projectedViewFinder.width())
+    // Creates a rect of a certain aspect ratio surrounding the view finder
+    val aspectRatioCrop = projectedViewFinderSquare.centerOn(projectedViewFinder).expandToAspectRatio(9, 16)
+    // Crops the image and fills the rest of the aspect-ratio image with gray
+    val croppedImage = this.cropWithFill(aspectRatioCrop, aspectRatioCrop.intersection(this.size().rect()))
+    // Finds the center rectangle of the newly cropped image
+    val originalCenterRect = originalCenterSize.centerOn(croppedImage.size().rect())
+    // Produces a map of rects to rects which are used to map segments of the old image onto the new one
+    val regionMap = croppedImage.resizeRegion(originalCenterRect, futureCenterRect, futureImageSize)
+    // construct the bitmap from the region map
+    return croppedImage.rearrangeBySegments(regionMap, futureImageSize)
+}
+
+/**
+ * Crops and image using originalImageRect and places it on finalImageRect, which is filled with
+ * gray for the best results
+ */
+fun Bitmap.cropWithFill(finalImageRect: Rect, originalImageRect: Rect): Bitmap {
```

Comment:
we don't need `originalImageRect` since we have a reference to the bitmap already in `this`